extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/ArcProxyBeanMetaDataClassNormalizer.java

```diff
@@ -20,6 +21,9 @@ public <T> Class<? super T> normalize(Class<T> beanClass) {
         while (Subclass.class.isAssignableFrom(targetClass)) {
             targetClass = targetClass.getSuperclass();
         }
+        while (ClientProxy.class.isAssignableFrom(targetClass)) {
+            targetClass = targetClass.getSuperclass();
+        }
```

Comment:
Are you absolutely sure the hierarchy is always original -> `ClientProxy` -> `SubClass`? Or do you expect that `SubClass` and `ClientProxy` will never be mixed in the same hierarchy?

This would seem safer to me, just in case they are mixed in an unexpected order:

```suggestion
        while (Subclass.class.isAssignableFrom(targetClass) || ClientProxy.class.isAssignableFrom(targetClass)) {
            targetClass = targetClass.getSuperclass();
        }
```

Reply:
A subclass should never extend the client proxy. And a client proxy should never extend the subclass.

Reply:
Okay, well if you're certain, let's go with that.

Reply:
FWIW, what @mkouba says was my understanding as well, thus why I implemented it like this.